README.md

```diff
@@ -128,19 +128,19 @@ You will need to understand the [**wire protocol specification**](docs/wire_prot
 The software is provided under Apache-2.0. Contributions to this project are accepted under the same license.
 
 This project uses the following third party crates:
-* serde (Apache-2.0)
+* serde (MIT and Apache-2.0)
```

Comment:
When the crate choose a double-licensing option (Apache-2.0 **OR** MIT), as the following (example `uuid`):

> # License
>
> Licensed under either of:
> * Apache License, Version 2.0, (LICENSE-APACHE or https://www.apache.org/licenses/LICENSE-2.0)
> * MIT license (LICENSE-MIT or https://opensource.org/licenses/MIT)
> at your option.
>
> ## Contribution
>
>Unless you explicitly state otherwise, any contribution intentionally submitted for inclusion in the work by you, as defined in the Apache-2.0 license, shall be dual licensed as above, without any additional terms or conditions.

We explicitely make a choice and choose in that file to use Apache-2.0 license. If it is **AND**, we keep the two licenses.

Reply:
I can revert the licensing change, but that sounds like a false dichotomy. All of those crates are dual-licensed, in the sense that you can pick either of them. The difference you noticed is in the manifest. People used to write `MIT/Apache-2.0`, but now `MIT OR Apache-2.0` is preferred because it's "more machine-readable". Both are equivalent, see the comment in https://doc.rust-lang.org/stable/cargo/reference/manifest.html#package-metadata.

Reply:
Thanks for the pointer. What I understand from that is that is is more explicit to use **AND** or **OR** instead of a slash (`/`) but not that they both mean the same thing 😛 
But I do not know if the licenses are for use of the library or contributions to it, and if it is for contributions that means that there could be files both licensed under MIT and Apache-2.0 in the same library.
In that case, it is probably better to use **AND** everywhere as you changed.

Reply:
I think `OR` would be a better fit.

>  But I do not know if the licenses are for use of the library or contributions to it, and if it is for contributions that means that there could be files both licensed under MIT and Apache-2.0 in the same library.

It's for the crate, and dual-licensing means that the user of the crate gets to pick one of the licenses (or both) that fits their project better. By contributing to a crate you implicitly admit that your changes will be licensed under the crate's terms (e.g. both MIT and Apache-2.0).

Reply:
Would not it feel weird to have **OR** on this file as we are building a binary and not re-distributing the crates? As in, we are the last one in the chain so should not we make that decision between Apache-2.0 or MIT?

Reply:
I think it makes sense to say "PARSEC is licensed under Apache-2.0 and uses the following dependencies, with their associated licenses", but I am not a lawyer :-).